File: uncategorized/1051_height_checker.ts
function heightChecker(heights: number[]): number {

    let sorted_heights = [... heights];
    sorted_heights.sort((a,b) => a-b);

    // console.log(sorted_heights, heights);
    let ans = 0;

    for(let i=0; i<heights.length; i++){
        if(sorted_heights[i] != heights[i]){
            ans += 1;
        }
    }

    return ans;

};

File: uncategorized/1051_height_checker.rs
impl Solution {
    pub fn height_checker(heights: Vec<i32>) -> i32 {

        let mut sorted_heights = heights.clone();
        sorted_heights.sort();

        let mut ans = 0;

        for i in 0.. heights.len() {
            if sorted_heights[i] != heights[i] {
                ans += 1;
            }
        }

        ans

    }
}